assets/js/components/GoogleChart/index.js

```diff
@@ -197,6 +205,74 @@ export default function GoogleChart( props ) {
 		}
 	}, [ onMouseOver, onMouseOut ] );
 
+	/**
+	 * Adds any "key date" vertical lines/tooltips to the charts.
+	 */
+	const addKeyDateLinesToChart = () => {
+		if ( ! chartWrapperRef.current ) {
+			return;
+		}
+
+		if ( ! dateMarkers.length ) {
+			return;
+		}
+
+		const chart = chartWrapperRef.current.getChart();
+		const chartLayoutInterface = chart?.getChartLayoutInterface();
+		const chartArea = chartLayoutInterface?.getChartAreaBoundingBox();
+		const dataTable = chartWrapperRef.current?.getDataTable();
+
+		if ( ! chartLayoutInterface || ! chartArea || ! dataTable ) {
+			return;
+		}
+
+		// Add the dotted line and tooltip for each date marker.
+		dateMarkers.forEach( ( dateMarker, index ) => {
+			const dateFromMarker = new Date( Date.parse( dateMarker.date ) );
+
+			const chartLine = document.querySelector(
+				`#googlesitekit-chart__date-marker-line--${ instanceID }-${ index }`
+			);
+			invariant(
+				chartLine,
+				`#googlesitekit-chart__date-marker-line--${ instanceID }-${ index } is missing from the DOM, but required to render date markers.`
+			);
+
+			// Align the dotted line with the date for this marker.
+			chartLine.style.left =
+				Math.floor(
+					chartLayoutInterface.getXLocation( dateFromMarker )
+				) -
+				1 +
+				'px';
+			chartLine.style.top = Math.floor( chartArea.top ) + 18 + 'px';
```

Comment:
The literal `18` is used in a few places and I can see it refers to the icon size. I'd suggest using a nicely named constant for this, for the sake of clarity.